### ISSUE TITLE
Iteration of GArray, GDictionary

### ISSUE THREAD

**Issue by GeTechG** [godotjs/GodotJS#19]
Maybe it's already there and I just haven't figured out how, hint then how it can be done. If not, it would be great to iterate arrays.
```js
const arr = new GArray(["one", "two"]);
for (let arrElement of arr) {
    console.log(arrElement);
}
```

**Comment by ialex32x** [comment]
The iteration support on Godot Array/Dictionary is implemented in the module `jsb.core`. So it works until `jsb.core` is initially loaded. You could explicitly call `require("jsb.core")` to ensure it's loaded.

Here is how it's implemented under the hood:
```ts
// the source code is in GodotJS/scripts/jsb.bundle/src/jsb.core.ts

(function (items: Array<{ class: any, func: Function }>) {
    for (let item of items) {
        item.class.prototype[Symbol.iterator] = item.func;
    }
})(
    [
        {
            class: GDictionary,
            func: function* () {
                let self: GDictionary = <any>this;
                let keys = self.keys();
                for (let i = 0; i < keys.size(); ++i) {
                    const key = keys.get_indexed(i);
                    yield { key: key, value: self.get_keyed(key) };
                }
            }
        },
        {
            class: GArray,
            func: function* () {
                let self: GArray = <any>this;
                for (let i = 0; i < self.size(); ++i) {
                    yield self.get_indexed(i);
                }
            }
        }
    ]
);
```

The current approach isn't very reasonable, and I'll work on improving it later. 
And directly using JS Array/Map may be more straightforward if there is no interaction with godot methods.

**Comment by ialex32x** [comment]
Here is an [example](https://github.com/godotjs/GodotJSExample/blob/main/tests/test_dictionary.ts) of how to iterate on GArray/GDictionary with the iterator implemented in `jsb.core`.

**Comment by GeTechG** [comment]
I tried to move the project under the new edits and faced with the fact that the code where I have iterables does not work, although `jsb.core` I imported, I tried to replace it with `godot.annotations` it did not help, tell me what I do wrong.

**Comment by ialex32x** [comment]
Please try the latest main branch (and regenerate all d.ts in your projects for a better dev experience), sorry for the confusion of changes in `jsb.core`, these iterator injections are now migrated to `jsb.inject`, and no explicit `require` is needed to use them anymore, they will be automatically injected once the GArray/GDictionary type is loaded.

**Comment by GeTechG** [comment]
> Please try the latest main branch (and regenerate all d.ts in your projects for a better dev experience), sorry for the confusion of changes in `jsb.core`, these iterator injections are now migrated to `jsb.inject`, and no explicit `require` is needed to use them anymore, they will be automatically injected once the GArray/GDictionary type is loaded.

I updated and I have `jsb.inject` missing from the declaration. It worked on 0.9.8 (2e98b410fdc53f53b3c2c8db790a8e57db287d34) commit.

**Comment by ialex32x** [comment]
> > Please try the latest main branch (and regenerate all d.ts in your projects for a better dev experience), sorry for the confusion of changes in `jsb.core`, these iterator injections are now migrated to `jsb.inject`, and no explicit `require` is needed to use them anymore, they will be automatically injected once the GArray/GDictionary type is loaded.
> 
> I updated and I have `jsb.inject` missing from the declaration. It worked on 0.9.8 ([2e98b41](https://github.com/godotjs/GodotJS/commit/2e98b410fdc53f53b3c2c8db790a8e57db287d34)) commit. I understand the problem is that it does not compile at all, because even after compiling ts in js I have errors that there is no iterator, so there is no import at all. I'm using v8 if it's important.

My bad, it should be fixed in the latest commit *57ff438*. I used wrong type names in `jsb.inject`.

**Comment by GeTechG** [comment]
> My bad, it should be fixed in the latest commit _[57ff438](https://github.com/godotjs/GodotJS/commit/57ff43829ef8e20f1bb897b6320a311513ba8dc4)_. I used wrong type names in `jsb.inject`.

Yes it's ok now, thanks, I understand the iterators are imported by themselves now?

**Comment by ialex32x** [comment]
Yes, they are available without additional `require` in user scripts.
